### ISSUE TITLE
Example XOR training overly optimistic

### ISSUE THREAD

**Issue by kichooo** [cazala/synaptic#93]
In the example on XOR training, results in the README are too optimistic.
```
myPerceptron.activate([0,0]); // 0.0268581547421616
myPerceptron.activate([1,0]); // 0.9829673642853368
myPerceptron.activate([0,1]); // 0.9831714267395621
myPerceptron.activate([1,1]); // 0.02128894618097928
```
Actual results:

```
[ 0.049756069386454695 ]
[ 0.9358857430205025 ]
[ 0.9360991915005301 ]
[ 0.08569924701231879 ]
```

Code:

```
const synaptic = require('synaptic'); // this line is not needed in the browser
const Neuron = synaptic.Neuron,
  Layer = synaptic.Layer,
  Network = synaptic.Network,
  Trainer = synaptic.Trainer,
  Architect = synaptic.Architect;

const myNetwork = new Architect.Perceptron(2, 3, 1);
const myTrainer = new Trainer(myNetwork);

myTrainer.XOR(); // { error: 0.004998819355993572, iterations: 21871, time: 356 }

console.log(myNetwork.activate([0, 0])); // 0.0268581547421616
console.log(myNetwork.activate([1, 0])); // 0.9829673642853368
console.log(myNetwork.activate([0, 1])); // 0.9831714267395621
console.log(myNetwork.activate([1, 1])); // 0.02128894618097928
```

It is misleading for the user, as you probably need other settings in place to achieve such results. New users might feel confused that their XOR trainer is worse.

**Comment by menduz** [comment]
That depends on the actual network weights, maybe the initial random set evolved on a local minimum at some point, and the given iterations + learning rate were not enough to reach the desired error.

You can try this: Execute your code several times and take a look at the results, those may vary a little bit

**Comment by kichooo** [comment]
Hey, thank you for you answer. I executed it multiple times, never getting out of 0.93-0.94 range

**Comment by cazala** [comment]
@kichooo that's because you are telling the network to stop when the error goes below `0.005` (actually you are not passing any threshold, and that is the default one)

Try this:

```javascript
const synaptic = require('synaptic'); // this line is not needed in the browser
const Neuron = synaptic.Neuron,
  Layer = synaptic.Layer,
  Network = synaptic.Network,
  Trainer = synaptic.Trainer,
  Architect = synaptic.Architect;

const myNetwork = new Architect.Perceptron(2, 3, 1);
const myTrainer = new Trainer(myNetwork);

myTrainer.XOR({ error: 0.001 })

console.log(myNetwork.activate([0, 0]));
console.log(myNetwork.activate([1, 0]));
console.log(myNetwork.activate([0, 1]));
console.log(myNetwork.activate([1, 1]));
```

☝️ notice that I passed `{ error: 0.001 }` to the `.XOR()`
